core/src/main/java/org/jdbi/v3/core/mapper/reflect/ReflectionMappers.java

```diff
@@ -44,9 +42,9 @@ public class ReflectionMappers implements JdbiConfig<ReflectionMappers> {
      * snake_case matching for names.
      */
     public ReflectionMappers() {
-        columnNameMatchers = Arrays.asList(
```

Comment:
I am not sure these change is needed at all. The list is never modified and the only place where it is copied is in the c'tors which are by definition single-threaded.

Reply:
The two constructors created two different `ArrayList`s with different behaviour.
`java.util.ArrayList != java.util.Arrays.ArrayList`
I prefer if such subtle differences are avoided.

Reply:
what is the different behavior? Where does it manifest to the end user. We give zero guarantees besides "you can get a list from this class". I would prefer us using `List.of` as those are explicitly called unmodifiable. Alas, that is Java 9...

Reply:
Much prefer using one way of creating an array list instead of two: avoid one import and improved readability
I would also prefer `List.of` ...
https://www.baeldung.com/java-arrays-aslist-vs-new-arraylist